jcl/src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -413,12 +413,19 @@ MethodType accessModeType(Class<?> receiver, Class<?> type, Class<?>... args) {
 		}
 
 		this.fieldType = setter.parameterType(setter.parameterCount() - 1);
-		this.coordinateTypes = getter.parameterArray();
+		
+		/* The first VarHandle parameter type should be removed from the getter in order to derive
+		 * the coordinate types. 
+		 */
+		Class<?>[] getterParams = getter.parameterArray();
+		this.coordinateTypes = Arrays.copyOfRange(getterParams, 1, getterParams.length);
```

Comment:
Is `getter` guaranteed to have at least 1 element at this point?

Reply:
Yes, I believe so. I expect atleast one parameter of type `VarHandle` for all `AccessMode` types.